### PR TITLE
Clarify getting_started without losing information

### PR DESCRIPTION
**Pull request by Blaisorblade**
To a user trying to get started with ENSIME, "ENSIME has a JVM process that indexes your dependencies and understands your source code using scalac and javac Abstract Syntax Trees." is not most helpful, because:
- he might not know what ASTs are.
- the text does not clarify why that information is relevant.

I've reordered the information and clarified a few things that are worth spelling out. Apart from the mention "Abstract Syntax Trees" I've tried to drop no important information; if that's important, maybe we can iterate on this version.

I don't think the new text is excellent, but it tries to be more accessible.

### CODE REVIEW

**Comment by fommil** [comment]
Hmm, I see what you are trying to do but it is too wordy: paragraphs should lead with key information and not be longer than two sentences.

It is also apologetic: please do not start a sentence with a word followed by a comma.

Therefore I do not feel it is an improvement in its current form. If you have a second draft, I would be happy to take a look.

Developers do not typically choose their build tool, this is enforced by their management.

If you feel that dropping the AST reference is important, let's start there and interate. I intentionally did this to scare off people who do not understand (or care to learn about) what an AST is. Remember, I am not interested in users: only contributors and sponsors. If you have a better way of doing that, I'm happy to hear since this is most likely an implicit bias.

Signing the CLA is a requirement.

**Comment by Blaisorblade** [comment]
I agree this is wordy and that paragraphs should lead with key info. I have side issues with some other advice, but let's first make sure our goals are compatible, and that I get the goals for your docs.

> I intentionally did this to scare off people who do not understand (or care to learn about) what an AST is. Remember, I am not interested in users: only contributors and sponsors.

I was trying to prevent scaring off people. I didn't put effort in the PR, because many don't believe this sort of thing can scare off people even after you point it out. I did not realize it was intentional O_O.

But this text contrasts with your stated goals — which is maybe why you're open to changing it? I've re-read The Open Source Entitlement Complex. I agree with the following quote:

> Libre projects could do a better job of being helpful, yet firm, with users who are starting out or experiencing problems. Better documentation is a good way to keep users from distracting developers. Users who can educate themselves are always welcome, even if they never contribute to or sponsor a project!

Since those "welcome users" don't actually need to learn what an AST is, that seems to be violating your principle. So we can stop scaring them off?

That paragraph also sounds like I am welcome to use Ensime without yet committing to future contributions?

> If you have a better way of doing that, I'm happy to hear since this is most likely an implicit bias.

I'm honestly confused: a better way of doing what?
And why do you talk about implicit bias—unless it's an implicitly bias against users, even ones doing nothing wrong?

**Comment by fommil** [comment]
I'm operating on the belief that a user who doesn't know, or care, (about ASTs) is going to demand more of my time. This may be a false assumption, but from the amount of time I had to spend explaining it in the gitter room, I have some evidence. That said, I am willing to remove this because I could be wrong.

Let's go back to the original text. Remove everything you do not feel is relevant. Then we'll see where clarifying text needs to be added. Then we can see if rearranging anything helps.

Try to keep bullet points to <10 words and paragraphs to two sentences: I have watched people read the website and if text does not appear short and snappy, especially in an intro page, they look for actionable links (some even reach for the search tool and type "install").

It is a very different medium to a technical paper: very few people want to read this text. Therefore I propose that it is only possible to install ensime by reading the docs and anybody who skips the docs is unable to do so. I've considered adding in a magic variable before ensime will start, to prove that you read the docs. Or a question on startup that proves you understand how to report problems (i.e. Troubleshooting).

**Comment by Blaisorblade** [comment]
> It is a very different medium to a technical paper: very few people want to read this text. Therefore I propose that it is only possible to install ensime by reading the docs and anybody who skips the docs is unable to do so. I've considered adding in a magic variable before ensime will start, to prove that you read the docs. Or a question on startup that proves you understand how to report problems (i.e. Troubleshooting).

Both here and for papers, readers won't care for info they're not convinced they need. So you tell them why they need info, and drop the rest. If you instead fight readers, they'll fight back, with the effects you report. (I'll concede paper readers might have a bit more patience, but less than you seem to think).

Better: to produce good text, blame the text and not the readers if they misunderstand it or skip it. Users (or people) are more diverse than one realizes. (With a few caveats, but less than you think). 

> I'm operating on the belief that a user who doesn't know, or care, (about ASTs) is going to demand more of my time.

Maybe yes, but you could give them arbitrary challenges that correlate statistically with relevant computer skills, even though the challenge is irrelevant to using ENSIME. Maybe something like ROT13'ing the docs would actually help.

Anyway, respectfully speaking, I'm now convinced our goals and ideas are not compatible, and I'll leave. That was my idea after https://github.com/ensime/ensime.github.io/issues/120; your blog post convinced me to reconsider, but this allowed me to confirm my conclusion, mostly because your behavior seems less welcoming to users than your post suggests.

With my ideas about good docs (which you probably disagree with), your stated plans are equivalent to intentionally crippling them, so the effects aren't surprising. It seems you spent more time on docs than I admitted, but apparently in the wrong direction.
I also retract my claim I'd be able to help. Whatever writing skill can't get user-friendly docs from a user-hostile project. (Which is why this PR was just a first revision and pretty lazy, I'd have spent more time with a different reaction).

You don't have any duty to do anything about this issue, but stop trying to shame users and reporters into fixing bugs of your project. Anybody can point out issues and do nothing about it, so other people who care about those issues won't try using Ensime and annoy you.
In fact, as long as some issues get fixed anyway, reporters can even hope (not demand!) to get them fixed, so please don't question that.

One last helpful tip I can offer: low-priority labels are much kinder ways to tell "we won't fix this", and I think they'll trigger less abuse. IMHO, keeping the bug tracker clean of low-priority issues is a goal that does more harm than good.

**Comment by fommil** [comment]
I think you're reading too much into the fact that I thought your proposed changes here are not better than what we have. Too wordy.

**Comment by fommil** [comment]
I did a quick pass of this page to remove text, in response to your suggestions. It probably does read better now. I honestly don't know how anybody could read this documentation site and claim that we are being "hostile" to users. In fact I hear from many people saying it is excellent and other projects have pointed to our documentation as something to aspire to.

I am sorry that your goals do not align with ensime or our philosophy, but there is nothing can be done about this.

**Comment by Blaisorblade** [comment]
> I did a quick pass of this page to remove superflous text, in response to some of your suggestions (but not the wordy additions). I think it reads better now.

Seems indeed better :-).

> I honestly don't know how anybody could read this documentation site and claim that we are being "hostile" to users.

It's about your comments. While I'm winding this down, if you're curious on my opinion (not that I'll try to convince you): people struggling have my sympathy, even when they struggle to understand. So I don't consider justified user hostility such as:

> I intentionally did this to scare off people who do not understand (or care to learn about) what an AST is.

It's at least good you undid that :-).

**Comment by fommil** [comment]
I am not really interested in helping people who don't want to learn... this is not hostile, it is purely nonchalant.

**Comment by SethTisue** [comment]
> I am not interested in users: only contributors and sponsors

if were you, I would be asking myself what the likelihood is of attracting contributors and sponsors to a project that doesn't have users. don't contributors and sponsors usually grow out of having a robust pool of users...?

**Comment by fommil** [comment]
@sethtisue we already have plenty of users. We need to increase the proportion of contributing/sponsoring existing users before we can grow any more. It's become too much of a burden on my time.

This is all moot because our docs are, in fact, very welcoming. The OP is the only person who feels that the docs are "hostile". I disagree.

I have not implemented anything to actively reduce the number of users coming on board... except to publicly speak out to remind whiny and abusive users that we owe them nothing.

My rant on the matter in concluded with the blog post, I plan on just ignoring such people going forwards. I don't believe I've done anything that any reasonable person wouldn't do.

**Comment by SethTisue** [comment]
> It's become too much of a burden on my time

hard choices all around, there :-\

**Comment by fommil** [comment]
To me clear, I mean dealing with rude, passive aggressive, and self-entitled users (who are unable to help themselves) are becoming the much of a burden on my time. Not maintaining and interesting ensime. In many ways, we're better than ever!
